src/hotspot/share/gc/shenandoah/shenandoahHeap.inline.hpp

```diff
@@ -686,6 +686,7 @@ size_t ShenandoahHeap::capture_old_usage(size_t old_usage) {
 }
 
 void ShenandoahHeap::set_previous_promotion(size_t promoted_bytes) {
+  shenandoah_assert_heaplocked();
```

Comment:
Is this related and/or relevant?

Reply:
We can remove that.  It was "somewhat" related to the historical progression that eventually resulted in this refactoring code.  At one point in my development, I had accidentally tried to set_previous_promotion() without having the heap lock, and the results were unpredictable.

Reply:
Hmm ok. No, leave it there. If it checks an important condition, then it should be useful.